app/cmd/process.go

```diff
@@ -202,11 +202,15 @@ func getProcessManagerClient(c *cli.Context) (*client.ProcessManagerClient, erro
 	tlsDir := c.GlobalString("tls-dir")
 
 	if tlsDir != "" {
-		return client.NewProcessManagerClientWithTLS(url,
+		imClient, err := client.NewProcessManagerClientWithTLS(url,
 			filepath.Join(tlsDir, "ca.crt"),
 			filepath.Join(tlsDir, "tls.crt"),
 			filepath.Join(tlsDir, "tls.key"),
 			"longhorn-backend.longhorn-system")
+		if err == nil {
+			return imClient, err
+		}
+		logrus.WithError(err).Info("Falling back to non tls client")
 	}
 
 	return client.NewProcessManagerClient(url, nil)
```

Comment:
NIT: Can we directly modify this function and `NewProcessManagerClientWithTLS` so that the caller does not need to handle the fallback by itself?

Reply:
Feel this is on purpose to make `NewProcessManagerClientWithTLS` and `NewProcessManagerClient` do their responsibilities precisely, but whether fallback or not leaves to the caller instead. 

This is more about responsibility segregation.

Reply:
Make sense. Thank you for your explanation.